From eb97b4969c66ca0fa9e91c339c7dc409cb6a9143 Mon Sep 17 00:00:00 2001
From: Tobias Ribizel <mail@ribizel.de>
Date: Wed, 21 Aug 2024 15:38:17 +0200
Subject: [PATCH] remove assertion workaround

This causes some kernels on ROCm debug builds to fail
---
 include/ginkgo/core/base/types.hpp | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/include/ginkgo/core/base/types.hpp b/include/ginkgo/core/base/types.hpp
index 8e2096c09e2..e375da15f9c 100644
--- a/include/ginkgo/core/base/types.hpp
+++ b/include/ginkgo/core/base/types.hpp
@@ -51,30 +51,8 @@
 #endif
 
 
-#if (defined(__CUDA_ARCH__) && defined(__APPLE__)) || \
-    defined(__HIP_DEVICE_COMPILE__)
-
-#ifdef NDEBUG
-#define GKO_ASSERT(condition) ((void)0)
-#else  // NDEBUG
-// Poor man's assertions on GPUs for MACs. They won't terminate the program
-// but will at least print something on the screen
-#define GKO_ASSERT(condition)                                               \
-    ((condition)                                                            \
-         ? ((void)0)                                                        \
-         : ((void)printf("%s: %d: %s: Assertion `" #condition "' failed\n", \
-                         __FILE__, __LINE__, __func__)))
-#endif  // NDEBUG
-
-#else  // (defined(__CUDA_ARCH__) && defined(__APPLE__)) ||
-       // defined(__HIP_DEVICE_COMPILE__)
-
-// Handle assertions normally on other systems
 #define GKO_ASSERT(condition) assert(condition)
 
-#endif  // (defined(__CUDA_ARCH__) && defined(__APPLE__)) ||
-        // defined(__HIP_DEVICE_COMPILE__)
-
 
 // Handle deprecated notices correctly on different systems
 // clang-format off
